File: app/src/main/java/com/vereshchagin/nikolay/stankinschedule/ui/settings/view/widget/SettingsWidgetHolder.java
package com.vereshchagin.nikolay.stankinschedule.ui.settings.view.widget;

import android.view.View;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.vereshchagin.nikolay.stankinschedule.R;

/**
 * Holder элемента RecyclerView с виджетами.
 */
class SettingsWidgetHolder extends RecyclerView.ViewHolder implements View.OnClickListener {

    private final SettingsWidgetAdapter.OnWidgetClickListener listener;
    private final TextView mScheduleTitle;
    private int mWidgetID;

    SettingsWidgetHolder(@NonNull View itemView, SettingsWidgetAdapter.OnWidgetClickListener listener) {
        super(itemView);
        this.listener = listener;

        mScheduleTitle = itemView.findViewById(R.id.widget_schedule_name);
        itemView.findViewById(R.id.widget_item).setOnClickListener(this);
    }

    /**
     * Обновляет данные в holder.
     *
     * @param scheduleName название расписания.
     * @param widgetID     ID виджета расписания.
     */
    void bind(@NonNull String scheduleName, int widgetID) {
        mScheduleTitle.setText(scheduleName);
        mWidgetID = widgetID;
    }

    @Override
    public void onClick(View v) {
        if (v.getId() == R.id.widget_item) {
            listener.OnWidgetClicked(mWidgetID);
        }
    }
}

File: legacy/src/main/java/com/vereshchagin/nikolay/stankinschedule/ui/settings/view/widget/paging/SettingsWidgetHolder.kt
package com.vereshchagin.nikolay.stankinschedule.ui.settings.view.widget.paging

import android.appwidget.AppWidgetManager
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.vereshchagin.nikolay.stankinschedule.databinding.ItemWidgetSettingsBinding

/**
 * Holder элемента RecyclerView с виджетами.
 */
class SettingsWidgetHolder(
    private val binding: ItemWidgetSettingsBinding,
    private val clickListener: (widgetId: Int) -> Unit,
) : RecyclerView.ViewHolder(binding.root) {

    /**
     * Текущий ID виджета.
     */
    private var widgetId = AppWidgetManager.INVALID_APPWIDGET_ID

    init {
        binding.widgetItem.setOnClickListener {
            clickListener(widgetId)
        }
    }

    /**
     * Обновляет данные в holder.
     * @param item данные виджета с расписанием.
     */
    fun bind(item: SettingsWidgetItem) {
        binding.widgetScheduleName.text = item.scheduleName
        this.widgetId = item.widgetId
    }

    companion object {
        /**
         * Создает SettingsWidgetHolder виджета расписания для отображения в списке.
         */
        fun create(
            parent: ViewGroup,
            clickListener: (widgetId: Int) -> Unit,
        ): SettingsWidgetHolder {
            return SettingsWidgetHolder(
                ItemWidgetSettingsBinding.inflate(
                    LayoutInflater.from(parent.context), parent, false
                ),
                clickListener
            )
        }
    }
}